src/blob.jl

```diff
@@ -95,6 +95,13 @@ end
     end
 end
 
+@inline function Base.getindex(blob::Blob{T}, i::Int) where {T}
+    @boundscheck if i < 1 || i > fieldcount(T)
+        throw(BoundsError(blob, i))
+    end
+    return Blob{fieldtype(T, i)}(blob + Blobs.blob_offset(T, i))
+end
```

Comment:
Cool, seems reasonable! :)

My first impression was that this should actually dereference the field as well, so you'd just have 

```julia
@test b[] == (1, 2)
@test b[1] == 1
@test b[2] == 2
```
And then instead of assigning values via `b[1][] = 1`, you could implement `setindex!()` on the Blob{Tuple} for indices, so you'd have:
```julia
b = Blobs.malloc(Tuple{Int,Int})
b[1] = 1
b[2] = 2
@test b[] == (1, 2)
```

This seems a closer match to what dereferencing/indexing means in C/C++, which seems like what this package is trying to mimic.

--------------

But then I guess if you actually _wanted_ to get a blob of a field (what's currently returned by `b[1]`), there still wouldn't be a good syntax for that? In C/C++, i think you'd write this as `&b[1]`, but i don't think we have that luxury. You could do something like `blob_field(b, 1)`, instead? That's maybe less elegant though.

Interested in your thoughts! What do you think about the two options?

Reply:
Great point! My example doesn't properly illustrate it, but yeah, I actually wrote this because I needed to get the blob of a field. Although... thinking about it, I think your syntax would also do the job for me!

That being said, this seems closer to other syntax that we already have. i.e. `blob.field` also returns the blob of `field` in `blob`. e.g.:

```
struct Foo
    bar::Int
end

foo_blob = Blobs.malloc(Foo)
bar_blob = foo_blob.bar

@assert bar_blob isa Blob

bar_blob[] = 42
```

Reply:
gotcha. Hrmm makes sense. I guess in c/c++ the syntax would have you _first_ dereference it, _then_ access the field (either `(*foo_blob).bar` or `foo_blob->bar`).

And the C/C++ syntax for what we're doing above is `(&(foo_blob->bar))`.

🤔 I see why we did it the way we did, then.

👍 okay! Yeah i think you picked the best option, then! :) ✅

Reply:
Awesome, thanks!